src/test/clojure/clara/performance/test_compilation.clj

```diff
@@ -0,0 +1,154 @@
+(ns clara.performance.test-compilation
+  (:require [clojure.test :refer :all]
+            [clara.rules.compiler :as com]
+            [clara.rules :as r]
+            [clara.rules.durability.fressian :as dura])
+  (:import [java.io ByteArrayOutputStream ByteArrayInputStream]))
+
+(defn filter-fn
+  [seed-keyword]
+  (= seed-keyword (keyword (gensym))))
+
+(defn time-execution
+  [func]
+  (let [start (System/currentTimeMillis)
+        _ (func)
+        stop (System/currentTimeMillis)]
+    (- stop start)))
+
+(defn execute-tests
+  [func iterations]
+  (let [execution-times (for [_ (range iterations)]
+                          (time-execution func))
+        sum #(reduce + %)
+        mean (/ (sum execution-times) iterations)
+        std (->
+              (into []
+                    (comp
+                      (map #(- % mean))
+                      (map #(Math/pow (double %) 2.0)))
+                    execution-times)
+              sum
+              (/ iterations)
+              Math/sqrt)]
+    {:std (double std)
+     :mean (double mean)}))
+
+(defn run-performance-test
```

Comment:
I rolled my own here, rather than using a library like `criterium` because criterium makes assumptions based on the classes loaded and compiled. This lead to infinite looping.

Reply:
"because criterium makes assumptions based on the classes loaded and compiled" => Could you elaborate?  What sorts of assumptions?

Reply:
@WilliamParker as @mrrodriguez mentioned `criterium` does try to warm up for JIT, [here](https://github.com/michalmarczyk/criterium/blob/0.4.0/src/criterium/core.clj#L396-L435). Again as Mike says, it seems that `criterium` is looking for a stable state before running benchmarks. While ideal for most situations, compilation in this case will never reach a steady state. Effectively making the test a really expensive `while (true)` block.

I will probably put a comment on this function mentioning the reasoning.